src/toil/jobStores/awsJobStore.py

```diff
@@ -865,3 +905,131 @@ def parse_bool(s):
+        """
+        self.jobStore=jobStore
+        self.jobStoreFileID=jobStoreFileID
+        self.fileType=fileType
+        self.jobStoreID=None
+        self._version=None
+        self._buffer=None
+
+    @property
+    def bucket(self):
+        if self.fileType==FileInfo.typeRegular:
+            return self.jobStore.files
+        elif self.fileType==FileInfo.typeStats:
+            return self.jobStore.stats
+        else:
+            assert False
+    @property
+    def version(self):
+        return self._version
+    @version.setter
+    def version(self, version):
+        assert self._version is None and self._buffer is None
+        self._version = version
+    @property
+    def buffer(self):
+        return self._buffer
+    @buffer.setter
+    def buffer(self, buffer):
+        assert self._buffer is None and self._version is None
+        self._buffer = buffer
+
+    def toSDBItem(self, isProtected=True):
+        """
+        Transforms the buffer into an optionally encrypted SDB-storable item
+        :param isProtected: bool indicating whether this file is encrypted
+        :return:
+        """
+        item = {}
+        bytes_per_attr = 1024 - encryptionOverhead
+        if bool(isProtected and self.jobStore.sseKeyPath):
+            stringToEncode = encrypt(self.buffer,self.jobStore.sseKeyPath)
+        else:
+            stringToEncode = self.buffer
+        serializedAndEncodedJob = base64.b64encode(bz2.compress(stringToEncode))
+        # this convoluted expression splits the string into the max value for an attribute in SDB
+        jobChunks = [serializedAndEncodedJob[i:i + bytes_per_attr]
+                     for i in range(0, len(serializedAndEncodedJob), bytes_per_attr)]
+        for attributeOrder, chunk in enumerate(jobChunks):
+            item[str(attributeOrder).zfill(3)] = chunk
+        return item
+
+    def fromSDBItem(self, item, isProtected=True):
+        """
+        Transforms an optionally enctyped SDB item into it's original string form
+        :param sseKey: str encryption key
+        :param isProtected: bool indicating whether this file is encrypted
+        :return:
+        """
+        chunkedFile = item.items()
+        chunkedFile.sort()
+        if len(chunkedFile) == 1:
+            # First element of list = tuple, second element of tuple = serialized job
+            wholeFileString = chunkedFile[0][1]
+        else:
+            wholeFileString = ''.join(item[1] for item in filter(lambda x: x[0].isdigit(), chunkedFile))
+        decoded = bz2.decompress(base64.b64decode(wholeFileString))
+        if bool(isProtected and self.jobStore.sseKeyPath):
+            return decrypt(decoded,self.jobStore.sseKeyPath)
+        else:
+            return decoded
+
+    NoneString = ""
+    @classmethod
+    def load(cls, jobStore, jobStoreFileID, isProtected=True):
+        """
+        new object that is File or Version
+        :rtype: tuple(str version, AWS bucket)
```

Comment:
Don't invent new notation. Use :rtype: with PEP 0484 notation. Use :return: to describe return value in English.